packages/mgt-chat/src/components/ChatListItem/ChatListItem.tsx

```diff
@@ -195,19 +202,41 @@ export const ChatListItem = ({ chat, myId, onSelected }: IMgtChatListItemProps &
     return removeHtmlPTags(previewString);
   };
 
+  // if chat changes, update the internal state to match
+  useEffect(() => {
+    setChatInternal(chat);
+  }, [chat]);
+
+  // enrich the chat if necessary
+  useEffect(() => {
+    if (chatInternal.id && (!chatInternal.chatType || !chatInternal.members)) {
+      const provider = Providers.globalProvider;
+      if (provider && provider.state === ProviderState.SignedIn) {
+        const graph = provider.graph.forComponent('ChatListItem');
+        const load = async (id: string): Promise<Chat> => {
```

Comment:
Seemed like this code can be simplified.

```ts
const load = (id: string): Promise<Chat> => {
  return loadChatWithPreview(graph, id);
};
```

Reply:
yes, I think you are right.

Reply:
I was thinking the Chat state for ChatListItem should be managed externally by ChatList. I understand there is the idea that each component should be responsible for its own state. But if we consider ChatListItem to be an internal component and only used in context of ChatList, wouldn't be safe for its state to be managed by ChatList?